### PR TITLE
Add code upload

### PR DESCRIPTION
**Pull request by nikitayutanov**
Adds the ability to upload code

### CODE REVIEW

File: website/frontend/src/components/pages/Programs/children/Upload/children/CodeModal/CodeModal.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    const buffer = Buffer.from(arrayBuffer);
+    return api.code.submit(buffer);
+  };
+
+  const handleUpload = ({ events }: ISubmittableResult, hash: Hex) =>
+    events.forEach(({ event: { method } }) => {
+      if (method === 'CodeSaved') {
+        handleSuccess(hash);
+      } else if (method === 'ExtrinsicFailed') {
+        handleFail('ExtrinsicFailed');
```
Comment (osipov-mit) [inline website/frontend/src/components/pages/Programs/children/Upload/children/CodeModal/CodeModal.tsx:61]: Display text of errors here. After this https://github.com/gear-tech/gear-js/pull/390 will be merged

**Review by osipov-mit — CHANGES_REQUESTED** [review]

**Review by osipov-mit — APPROVED** [review]

File: website/frontend/src/components/pages/Programs/children/Upload/children/CodeModal/CodeModal.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    } catch (error) {
+      console.error(error);
+      handleFail('Something went wrong');
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  return (
+    <Modal
+      open={true}
```
Comment (codev0) [inline website/frontend/src/components/pages/Programs/children/Upload/children/CodeModal/CodeModal.tsx:104]: is this modal should open everytime?

**Review by aderrod94 — APPROVED** [review]

File: website/frontend/src/components/pages/Programs/children/Upload/children/DropTarget/DropTarget.tsx [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    },
+    [setDroppedFile, type]
+  );
+
+  const hiddenFileInput = React.useRef(null);
+
+  const handleClick = () => {
+    if (hiddenFileInput !== null && hiddenFileInput.current !== null) {
+      // @ts-ignore
+      hiddenFileInput.current.click();
```
Comment (codev0) [inline website/frontend/src/components/pages/Programs/children/Upload/children/DropTarget/DropTarget.tsx:51]: could you try to redo this with optional chaining?

**Review by EugenWay — APPROVED** [review]

File: website/frontend/src/components/pages/Programs/children/Upload/children/CodeModal/CodeModal.tsx (hunk 2 above)
Comment (nikitayutanov) [reply website/frontend/src/components/pages/Programs/children/Upload/children/CodeModal/CodeModal.tsx:104]: Yep, we control it's visibility in the parent component. Modal would be refactored anyway and `open` prop should be gone.

**Pull request merged**